src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js

```diff
@@ -45,6 +48,9 @@ const Section = Flyout.Section;
 // A counter for creating unique keys per new condition
 let conditionKey = 0;
 
+// A counter for creating unique keys per new action
+let actionKey = 0;
```

Comment:
We don't need this. We can make use of ComponentArray component. I'll comment below on how to do it.

Reply:
this is how conditions is done on this page so I'm going to leave it for now. We can improve it later